runtime/doc/lsp.txt

```diff
@@ -667,10 +667,29 @@ LspProgress                                                       *LspProgress*
     properties, where `params` is the request params sent by the server (see
     `lsp.ProgressParams`).
 
-    Example: >vim
+    Examples:
+
+    Redraw the statusline whenever an LSP progress message arrives: >vim
         autocmd LspProgress * redrawstatus
 <
 
+    Tell the parent terminal to indicate progress using a native progress
+    indicator (requires a supporting terminal): >lua
+
+        vim.api.nvim_create_autocmd('LspProgress', {
+          callback = function(ev)
+            local value = ev.data.params.value
+            if value.kind == 'begin' then
+              io.stdout:write('\027]9;4;1;0\027\\')
```

Comment:
Is it not possible that user code writing to `io.stdout` could interfere with Nvim's UI?

Reply:
Sure but the question still stands. Arbitrarily writing to the same channel that the UI itself uses, seems questionable.

Reply:
> Arbitrarily writing to the same channel that the UI itself uses, seems questionable.

In the TUI `io.stdout` is the TTY. We're writing control sequences to the terminal, so we have to write to the same channel as the UI.

Reply:
Yes. The question is could they overlap/intefere with stuff the TUI itself is doing? Should plugin/user code that emits term sequences be plumbed through the TUI somehow?

Reply:
Ah I see your point.

> Should plugin/user code that emits term sequences be plumbed through the TUI (tui.c) somehow? (E.g. nvim_ui_write)

I think this makes sense.